src/pudl/output/pudltabl.py

```diff
@@ -718,8 +731,9 @@ def hr_by_unit(self, update=False):
 
         """
         if update or self._dfs['hr_by_unit'] is None:
-            self._dfs['hr_by_unit'] = pudl.analysis.mcoe.heat_rate_by_unit(
-                self)
+            self._dfs['hr_by_unit'] = (
+                pudl.analysis.mcoe.heat_rate_by_unit(self)
```

Comment:
Passing `self` to some other method sounds like object-oriented design gone wrong. I see that  this is not a problem introduced by this PR so I do not expect solution for it here but we should probably consider refactoring the code and eliminate this pattern in the future.

Reply:
Oh yes, the output/analysis layer needs a big overhaul. They were made originally in the very early days. I have a few ideas for this, but it'd be great to get your input on that when the time comes!